src/task.c

```diff
@@ -799,58 +829,46 @@ task_config_valid(uint64 num_bg_threads[NUM_TASK_TYPES])
  * -----------------------------------------------------------------------------
  */
 platform_status
-task_system_create(platform_heap_id    hid,
-                   platform_io_handle *ioh,
-                   task_system       **system,
-                   bool                use_stats,
-                   uint64              num_bg_threads[NUM_TASK_TYPES],
-                   uint64              scratch_size)
-{
-   platform_status rc = task_config_valid(num_bg_threads);
+task_system_create(platform_heap_id          hid,
+                   platform_io_handle       *ioh,
+                   task_system             **system,
+                   const task_system_config *cfg)
+{
+   platform_status rc = task_config_valid(cfg);
```

Comment:
Not sure how best to resolve this duplication of cfg-validation checks but here's my thought:

Even if you add a call to `task_config_valid()` in `task_system_config_init()`, I think it may still be ok to retain this call on L837. It's a small function doing a quick check.

It will cover the odd case if some test / other code were to call `task_system_create()` with a `cfg` which would fail the checks in this validator function.